generators/csharp/codegen/src/ast/ClassReference.ts

```diff
@@ -26,8 +26,16 @@ export class ClassReference extends AstNode {
     }
 
     public write(writer: Writer): void {
-        writer.addReference(this);
-        writer.write(`${this.name}`);
+        const qualifiedTypeNameRequired =
+            writer.getAllBaseNamespaces().has(this.name) &&
+            // qualified type name not required if we're in the same base namespace as the conflicting type
+            !writer.getNamespace()?.startsWith(`${writer.getRootNamespace()}.${this.name}`);
```

Comment:
can we move this into a helper function and specifically document the edge cases